### PR TITLE
Feat(installation): installation definition and event

### PR DESCRIPTION
**Pull request by marcoscaceres**
This is pretty rough still, but  might as well start getting feedback...

### CODE REVIEW

**Comment by marcoscaceres** [comment]
This contains a bunch of grammatical and spelling errors - but I'm more interested in the overall design. I'm sure that @kenchris will rip through this, but @anssiko could you check this also please?

**Comment by marcoscaceres** [comment]
@kenchris, I've got rid of the beforeinstallprompt stuff, so we can move forward with just the install event. I shadowed the WIP stuff for beforeinstallprompt as part of this commit so I can get to it later... yes, I should have branched it out, but whatever... lazy.

File: index.html [hunk 1]
```diff
@@ -189,46 +189,130 @@
...
-        Requirements for Installable Web Apps</a></cite> .
-      </p>
-      <p>
-        A web application is <dfn data-lt="installing">installed</dfn> if the
-        user agent has provided the end-user with a means of instantiating a
-        new <a>top-level browsing context</a> that has the manifest's members
-        applied to it. That is, the manifest's members, or their defaults, are
-        in effect on the <a>top-level browsing context</a> (see <a>application
-        context</a>).
+        A common use of a manifest is for a user agent to <dfn data-lt=
```
Comment (kenchris) [inline index.html]: use case?

a common use-case of a manifest is for a UA to allow installing the web application

File: index.html [hunk 2]
```diff
@@ -189,46 +189,130 @@
...
+        "installing|installed|installation">install</dfn> a web application:
```
Comment (kenchris) [inline index.html]: I think you want , or ; instaed of :

File: index.html [hunk 3]
```diff
@@ -189,46 +189,130 @@
...
+        whereby the user agent provides the end-user with a means of
+        instantiating a new <a>top-level browsing context</a> that has the
+        manifest's members applied to it. That is, the manifest's members, or
+        their defaults, are in effect on the <a>top-level browsing context</a>.
+        This distinguishes an installed web application from a traditional
+        bookmark, as opening a web page from a traditional bookmark will not
+        have a manifest <a>applied</a> to it.
```
Comment (kenchris) [inline index.html]: the manifest properties applied to it?

File: index.html [hunk 4]
```diff
@@ -189,46 +189,130 @@
...
-        An example of <a>installing</a> would be a user agent that allows the
-        end-user to add a shortcut to a web application on their device's
-        homescreen (using the name and one of the icons found in the manifest).
-        Then, when the end-user launches a web application through this
-        explicit means, the manifest is applied to the browsing context prior
-        to the <a>start URL</a> being loaded. This gives the user agent time to
-        apply the relevant values of the manifest, possibly changing the
-        display mode and screen orientation of the web application.
+        An example of a user agent that allows installation would be one that
+        provides the end-user with a means to add a shortcut to a web
```
Comment (kenchris) [inline index.html]: not sure if 'shortcut' is the used word for this today.

File: index.html [hunk 5]
```diff
@@ -189,46 +189,130 @@
...
+        application on their device's homescreen (using the name and one of the
```
Comment (kenchris) [inline index.html]: if could also well be to their launcher. Why not say that the application becomes part of the platforms normal way of launching applications, such as home screen, launcher, start menu etc

File: index.html [hunk 6]
```diff
@@ -189,46 +189,130 @@
...
+        install the application into a list of bookmarks within the user agent
+        itself.
       </p>
       <p>
-        Alternatively, an application context can be launched through a <a>deep
-        link</a> (a URL that is within scope of the installed web application);
-        in which case, the manifest is applied and the deep link is loaded
-        within the context of web application.
+        An end-user can <dfn data-lt="manual installation">manually</dfn>
+        trigger an <a>installation process</a> through the browser's
```
Comment (kenchris) [inline index.html]: s/an/the?

File: index.html (hunk 2 above)
Comment (marcoscaceres) [inline index.html]: went with ";"

File: index.html (hunk 4 above)
Comment (marcoscaceres) [inline index.html]: Went with "link"

**Comment by anssiko** [comment]
@marcoscaceres Much thanks for the PR! It looks awesome. Sorry for being late with the review. I submitted a PR #461 with my review comments. Ready to be integrated.

**Pull request merged**